R/tbl_hierarchical_rate_by_grade.R

```diff
@@ -0,0 +1,468 @@
+#' AE Rates by Highest Toxicity Grade
+#'
+#' @description
+#'
+#' A wrapper function for [gtsummary::tbl_hierarchical()] to calculate rates of highest toxicity grades with the options
+#' to add rows for grade groups and additional summary sections at each variable level.
+#'
+#' Only the highest grade level recorded for each subject will be analyzed. Prior to running the function, ensure that
+#' the toxicity grade variable (`grade`) is a factor variable, with factor levels ordered lowest to highest.
+#'
+#' Grades will appear in rows in the order of the factor levels given, with each grade group appearing prior to the
+#' first level in its group.
+#'
+#' @inheritParams gtsummary::tbl_hierarchical
+#' @inheritParams gtsummary::sort_hierarchical
+#' @inheritParams gtsummary::add_overall.tbl_hierarchical
+#' @param variables ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   A character vector or tidy-selector of 3 columns in `data` specifying a system organ class variable,
+#'   an adverse event terms variable, and a toxicity grade level variable, respectively.
+#' @param include_overall ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   Variables from `variables` for which an overall section at that hierarchy level should be computed.
+#'   An overall section at the SOC variable level will have label `"- Any adverse events -"`. An overall section at
+#'   the AE term variable level will have label `"- Overall -"`. If the grade level variable is included it has no
+#'   effect. The default is `everything()`.
+#' @param filter (`expression`)\cr
+#'   An expression that is used to filter rows of the table. See the Details section below for more information.
+#' @param grade_groups (`named list`)\cr
+#'   A named list of grade groups for which rates should be calculated. Grade groups must be mutually exclusive, i.e.
+#'   each grade cannot be assigned to more than one grade group. Each grade group must be specified in the list as a
+#'   character vector of the grades included in the grade group, named with the corresponding name of the grade group,
+#'   e.g. `"Grade 1-2" = c("1", "2")`.
+#' @param grades_exclude (`character`)\cr
+#'   A vector of grades to omit individual rows for when printing the table. These grades will still be used when
+#'   computing overall totals and grade group totals. For example, to avoid duplication, if a grade group is defined as
+#'   `"Grade 5" = "5"`, the individual rows corresponding to grade 5 can be excluded by setting `grades_exclude = "5"`.
+#' @param keep_zero_rows (`logical`)\cr
+#'   Whether rows containing zero rates across all columns should be kept. If `FALSE`, this filter will be applied
+#'   prior to any filters specified via the `filter` argument. Defaults to `FALSE`.
+#' @param x (`tbl_hierarchical_rate_by_grade`)\cr
+#'   A gtsummary table of class `'tbl_hierarchical_rate_by_grade'`.
+#'
+#' @details
```

Comment:
I think a couple of examples here would be helpful?

Reply:
Hi @ddsjoberg,
Regarding filtering by other variables (as is done in some STREAM variants), we can add that capability within `cards::filter_ard_hierarchical()` if you think that's something we should enable. Since this function just calls `filter_hierarchical()` internally during an intermediate step any updates to the main filtering function will apply here as well.

Since there is no function-specific filtering, what sort of examples did you have in mind? I added one more basic example to the Examples section.

Reply:
So for this table, we can only filter based on the grades, not the rates of AEs, right? Maybe we should have a call to discuss and decide how to proceed.....

Reply:
This is enabled now after the filtering updates from cards.